dashboard/app/controllers/api/v1/schools_controller.rb

```diff
@@ -7,4 +7,14 @@ def index
     end
     render json: serialized_schools
   end
+
+  # GET /dashboardapi/v1/schoolsearch/:q/:limit
+  def search
+    limit = [40, Integer(params[:limit])].min
+    schools = School.where("name LIKE ?", "%#{params[:q]}%").order(:name).limit(limit)
```

Comment:
Backing up a step, this API is to support auto-completion in our school dropdowns? If so, is the DB data stored in a manner to facilitate this? For example, this LIKE query is an unindexed query (there isn't any kind of index on `name`). That said, perhaps the table is sufficiently small (103K) that indexing matters less.